### PR TITLE
Update cometbft-db Go version to 1.22.5

### PR DESCRIPTION
**Pull request by alesforz**
This PR bumps the go version to 1.22.5.

### CODE REVIEW

**Comment by cason** [comment]
Is this not a problem for the main CometBFT repo? I think we are using go 1.22.2 there, last time I've checked.

**Comment by alesforz** [comment]
> Is this not a problem for the main CometBFT repo? I think we are using go 1.22.2 there, last time I've checked.

Yes, I opened a draft PR in which I'm bumping the go version of cometbft as well.

File: .changelog/v0.13.0/summary.md [hunk 1]
```diff
@@ -3,5 +3,6 @@
```
Comment (melekes) [inline .changelog/v0.13.0/summary.md:3]: Can we change the summary in a release PR (which actually produces the changelog and bumps version)?

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**